pimd/pim_iface.c

```diff
@@ -116,6 +119,26 @@ struct pim_interface *pim_if_new(struct interface *ifp, bool igmp, bool pim,
 	zassert(ifp);
 	zassert(!ifp->info);
 
+	if (ifp->ifindex == IFINDEX_INTERNAL) {
+		if (strcmp(ifp->name, "pimreg") == 0) {
+			if (PIM_DEBUG_PIM_TRACE) {
+				zlog_debug(
+					"%s: interface %s keeps ifindex IFINDEX_INTERNAL=%d, matching PIM_OIF_PIM_REGISTER_VIF=%d",
+					__PRETTY_FUNCTION__, ifp->name,
+					IFINDEX_INTERNAL,
+					PIM_OIF_PIM_REGISTER_VIF);
+			}
+		} else {
+			if_set_index(ifp, if_nametoindex(ifp->name));
```

Comment:
This breaks several assumptions about how we gather data in FRR for protocols.  interface data must come from zebra.  This is not the correct way to handle this situation and pim must intelligently handle this differently.  What exactly are you fixing here?